go/mysql/conn.go

```diff
@@ -1707,7 +1707,15 @@ func (c *Conn) IsMarkedForClose() bool {
 	return c.closing
 }
 
+func (c *Conn) IsShuttingDown() bool {
+	return c.listener.shutdown.Load()
+}
+
 // GetTestConn returns a conn for testing purpose only.
 func GetTestConn() *Conn {
 	return newConn(testConn{})
 }
+
+func GetTestServerConn(listener *Listener) *Conn {
+	return newServerConn(testConn{}, listener)
+}
```

Comment:
Can this function (and `GetTestConn` above it) be moved to `conn_flaky_test.go` instead? So it's in a proper `_test.go` file?

Reply:
Yes functions in test files can only be used locally. They aren't exported.

From the Internet - 
```
Files matching *_test.go are only compiled when testing the package they are part of. If you are testing package A that uses package B, you won't have access to the _test.go code from package B.
```